DetectiCam.Core/ResultProcessor/MqttPublisher.cs

```diff
@@ -0,0 +1,104 @@
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OpenCvSharp;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using uPLibrary.Networking.M2Mqtt;
+using uPLibrary.Networking.M2Mqtt.Messages;
+
+namespace DetectiCam.Core.ResultProcessor
+{
+    public class MqttPublisher : IAsyncSingleResultProcessor
+    {
+        private readonly ILogger _logger;
+        private readonly MqttClient? _client;
+        private readonly MqttPublisherOptions? _config;
+        private readonly bool _isEnabled = false;
+        private readonly string? _topicPrefix;
+
+        public MqttPublisher(   IOptions<MqttPublisherOptions> config,
+                                ILogger<MqttPublisher> logger)
+        {
+            if (config is null) throw new ArgumentNullException(nameof(config));
+            if (logger is null) throw new ArgumentNullException(nameof(logger));
+
+            _logger = logger;
+
+            try
+            {
+                _config = config.Value;
+                _isEnabled = _config.Enabled;
+
+                if (_isEnabled)
+                {
+                    if (!String.IsNullOrEmpty(_config.TopicPrefix))
+                    {
+                        _topicPrefix = _config.TopicPrefix;
+                        if (!_config.TopicPrefix.EndsWith("/", StringComparison.OrdinalIgnoreCase))
+                        {
+                            _topicPrefix += "/";
+                        }
+                    }
+                    else
+                    {
+                        _topicPrefix = "";
+                    }
+
+                    // create client instance 
+                    _client = new MqttClient(_config.Server, _config.Port, false,
+                        MqttSslProtocols.None,
+                        null, null);
+
+                    string clientId = String.IsNullOrEmpty(_config.ClientId) ? Guid.NewGuid().ToString() :
+                        _config.ClientId;
+
+                    _client.Connect(clientId, _config.Username, _config.Password);
+                }
+            }
+            catch (OptionsValidationException ex)
+            {
+                foreach (var failure in ex.Failures)
+                {
+                    _logger.LogError(failure);
+                }
+            }
+        }
+
+        public Task ProcessResultAsync(VideoFrame frame, DnnDetectedObject[] results)
+        {
+            if (_isEnabled && _client != null)
+            {
+
+                if (frame is null) throw new ArgumentNullException(nameof(frame));
+                if (results is null) throw new ArgumentNullException(nameof(results));
+
+                string strValue = "{ \"dectection\" : true }";
```

Comment:
Fix the typo: `dectection` -> `detection`

Reply:
Fixed in https://github.com/RaimondB/Detecticam/pull/4/commits/0e989a73d554c2a973b0e3a22dc31e5226c465da